pkgs/by-name/pt/ptouch-driver/package.nix

```
@@ -0,0 +1,91 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cups
+, cups-filters
+, foomatic-db
+, foomatic-db-engine
+, fetchpatch
+, ghostscript
+, libpng
+, libxml2
+, autoconf
+, automake
+, perl
+, patchPpdFilesHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ptouch-driver";
+  version = "1.7";
+
+  src = fetchFromGitHub {
+    owner = "philpem";
+    repo = "printer-driver-ptouch";
+    rev = "v1.7";
+    hash = "sha256-3ZotSHn7lERp53hAzx47Ct/k565rEoensCcltwX/Xls=";
+  };
+
+  patches = [
+    # Fixes some invalid XML file that otherwise causes a build failure
+    (fetchpatch {
+      name = "fix-brother-ql-600.xml";
+      url = "https://github.com/philpem/printer-driver-ptouch/commit/b3c53b3bc4dd98ed172f2c79405c7c09b3b3836a.patch";
+      hash = "sha256-y5bHKFeRXx8Wdl1++l4QNGgiY41LY5uzrRdOlaZyF9I=";
+    })
+  ];
+
+  buildInputs = [ cups cups-filters foomatic-db foomatic-db-engine ghostscript libpng libxml2 ];
```

```suggestion
  buildInputs = [ cups ghostscript libpng libxml2 patchPpdFilesHook ];
```

`cups-filters` (see below tho), `foomatic-db` and `foomatic-db-engine` are not needed at runtime, AFAICS? Or is there some non-obvious runtime dependency there?

Also, `patchPpdFilesHook` needs be included here for the hook to be actually executed.